### PR TITLE
pegtl: use modern cmake syntax and update license

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
- relates to https://github.com/taocpp/PEGTL/commit/c7630f1649906daf08b8ddca1420e66b542bae2b

**Review by branchv — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

File: audit_exceptions/permitted_formula_license_mismatches.json [hunk 1]
```diff
@@ -1 +1,3 @@
-[]
+[
+  "pegtl"
```
Comment (cho-m) [inline audit_exceptions/permitted_formula_license_mismatches.json:2]: Does our audit not work? We run the check on a specific tag so it should automatically fail when tag uses a new license
```console
❯ gh api '/repos/taocpp/PEGTL/license?ref=3.2.7' | jq .license.spdx_id
"MIT"

❯ gh api '/repos/taocpp/PEGTL/license?ref=c7630f1' | jq .license.spdx_id
"BSL-1.0"
```

File: Formula/p/pegtl.rb [hunk 2]
```diff
@@ -15,13 +16,18 @@ class Pegtl < Formula
...
   def install
-    mkdir "build" do
-      system "cmake", "..", *std_cmake_args,
-                            "-DPEGTL_BUILD_TESTS=OFF",
-                            "-DPEGTL_BUILD_EXAMPLES=OFF",
-                            "-DCMAKE_CXX_STANDARD=17"
-      system "make", "install"
-    end
+    odie "update license to BSL-1.0" if version > "3.2.7"
+
```
Comment (cho-m) [inline Formula/p/pegtl.rb:20]:
```suggestion
```

Similarly, I don't see any reason for this. Just let the audit automatically fail by not adding to exception list.

**Review by MikeMcQuaid — COMMENTED** [review]
Agreed with @cho-m

**Comment by chenrui333** [comment]
addressed the pr feedback.

**Review by carlocab — APPROVED** [review]

**Pull request merged**